warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IteratorBuildingVisitor.java

```diff
@@ -1058,21 +1058,22 @@ public Object visit(ASTNumberLiteral node, Object o) {
     private List<IvaratorCacheDir> getIvaratorCacheDirs() throws IOException {
         List<IvaratorCacheDir> pathAndFs = new ArrayList<>();
         
-        // first lets increment the count for a unique subdirectory
-        String subdirectory = ivaratorCacheSubDirPrefix + "term" + Integer.toString(++ivaratorCount);
+        StringBuilder subDirectoryBuilder = new StringBuilder();
+        subDirectoryBuilder.append(queryId);
+        if (scanId == null) {
+            log.warn("Running query iterator for " + queryId + " without a scan id.  This could cause ivarator directory conflicts.");
+        } else {
+            subDirectoryBuilder.append('_').append(scanId);
+        }
+        // and lets increment the count for a unique subdirectory
+        subDirectoryBuilder.append('_').append(ivaratorCacheSubDirPrefix).append("term").append(Integer.toString(++ivaratorCount));
```

Comment:
So the overall goal here is just to flatten out the directory structure?

Reply:
Yes